road/stats.py

```diff
@@ -0,0 +1,306 @@
+import os
+import pandas as pd
+import numpy as np
+import json
+import yaml
+
+
+def prepare_time_period_dfs(est_df, obs_df, times, combined_df_cols):
+    error = {col: est_df[col] - obs_df[col] for col in times}
+    error_df = pd.DataFrame(error)
+    error_squared_df = error_df.pow(2)
+    error_percent_df = error_df.divide(obs_df[times]).fillna(0)
+    error_percent_df = error_percent_df.apply(lambda x: x * 100, axis=1)
+
+    combined_df = pd.concat(
+        [est_df, obs_df[times], error_df, error_squared_df, error_percent_df], axis=1)
+    combined_df = combined_df.drop_duplicates(subset=['A', 'B'], keep='first')
+
+    calculation_cols = [
+        'Estimated Volume',
+        'Observed Volume',
+        'Errors',
+        'Squared Errors',
+        'Percent Errors']
+
+    time_period_dfs = {}
+
+    for period in times:
+        matching_column_indices = [
+            i for i, col_name in enumerate(
+                combined_df.columns) if col_name == period]
+        select_time_period_df = combined_df.iloc[:, matching_column_indices]
+
+        specific_columns_combined_df = combined_df[combined_df_cols]
+
+        select_time_period_loc_df = pd.concat(
+            [specific_columns_combined_df, select_time_period_df], axis=1)
+
+        new_column_names = combined_df_cols + \

```

Comment:
can we create the dataframe with these new columns directly, rather than creating the data and the column names separately? If something shifts / some changes happen in the future, and the data to column name correspondence is lost, it'll be super hard to trace this down